src/main.rs

```diff
@@ -305,10 +323,21 @@ impl HunkBuffer {
             diff_buffer,
             added_tokens,
             removed_tokens,
+            line_number_info,
             lines,
             config,
             stats,
         } = self;
+        let mut margin = [0u8; MAX_MARGIN];
+        let (mut current_line_minus, mut current_line_plus, margin) = match line_number_info {
+            Some(lni) => (
+                lni.minus_range.0,
+                lni.plus_range.0,
+                &mut margin[..cmp::max(MIN_MARGIN, lni.width())],
```

Comment:
minor: more idiomatic is `lni.width().max(MIN_MARGIN)`

Reply:
Thanks, I somehow didn't realize this was possible despite it appearing multiple times in the code base.